list.go

```diff
@@ -55,15 +55,23 @@ func (list *List) UnmarshalJSON(rawData []byte) error {
 MarshalJSON returns a top level object for the "data" attribute if a single object. In
 all other cases returns a JSON encoded list for "data".
 */
-func (list List) MarshalJSON() ([]byte, error) {
+func (list *List) MarshalJSON() ([]byte, error) {
 	// avoid stack overflow by using this subtype for marshaling
 	type MarshalList List
-	marshalList := MarshalList(list)
-	count := len(marshalList)
+	marshalList := (*MarshalList)(list)
+	isnil := marshalList == nil
```

Comment:
I'd rather just short circuit here on a `if marshalList == nil` statement and return nil, nil. Makes the logic that follows a bit simpler.